### PR TITLE
Galaxy travis testing

### PR DESCRIPTION
**Pull request by bgruening**
This includes the testing of Galaxy wrappers, very similar to what is deeptools is doing.

It is currently labeled as WIP, because hicPlotDistVsCounts seems to have a serious memory leak, which causes travis to kill the job.

### CODE REVIEW

**Comment by fidelram** [comment]
I will check this out

**Comment by fidelram** [comment]
The problem is that to run hicPlotDistVsCounts with high resolution data, the sparse matrix has to be converted into a dense matrix. This matrix is huge and does not fit into memory. I have some workarounds in mind but I can not implement them currently.

Best is to use this function only with low resolution data (say 10kb bins) but not with < 1kb bins. For this data the program works fine.

**Comment by fidelram** [comment]
@bgruening can you change the test matrix to the new low resolution matrix in `hicexplorer/test/test_data/small_test_matrix_50kb_res.h5`

File: galaxy/wrapper/hicFindTADs.xml [hunk 1]
```diff
@@ -76,39 +90,48 @@
...
             <param name="minDepth" value="20000"/>
             <param name="maxDepth" value="60000"/>
-            <param name="step" value="10000"/>
+            <param name="step" value="100000"/>
             <output name="outFileName" file="hicFindTADs_TAD_score.tabular" ftype="tabular"/>
+            <output name="tad_score_zscore_matrix" file="tadScoreFile.tabular_zscore_matrix.h5" ftype="h5" compare="sim_size"/>
         </test>
         <test>
             <param name="tadScoreFile" value="hicFindTADs_TAD_score.tabular" ftype="tabular"/>
+            <param name="tad_score_zscore_matrix" value="tadScoreFile.tabular_zscore_matrix.h5" ftype="h5"/>
```
Comment (bgruening) [inline galaxy/wrapper/hicFindTADs.xml]: @fidelram this file is 20MB in size so I don't check it into github. We might want to put all test data into a FTP storage and download them before running the tests? What do you think?

File: galaxy/wrapper/hicFindTADs.xml [hunk 2]
```diff
@@ -76,39 +90,48 @@
...
             <param name="mode_selector" value="find_TADs"/>
-            <param name="delta" value="0.001 0.01 0.03 0.05 0.1"/>
-            <param name="minBoundaryDistance" value="30000"/>
-            <param name="maxThreshold" value="0.2"/>
+            <param name="delta" value="0.002"/>
+            <param name="pvale" value="0.01"/>
+
             <output name="boundaries" file="hicFindTADs_find_boundaries.bed" ftype="bed"/>
             <output name="score" file="hicFindTADs_find_score.bedgraph" ftype="bedgraph"/>
             <output name="domains" file="hicFindTADs_find_domains.bed" ftype="bed"/>
```
Comment (bgruening) [inline galaxy/wrapper/hicFindTADs.xml]: The BED files and bedgraph file are empty in my testing with the latest version.

**Comment by bgruening** [comment]
@fidelram this branch contains a bunch of fixes and Galaxy wrapper testing. it is also an update to version 1.6.1.
I encountered a bug which needs a little bit discussion I think or a slightly bigger restructuring of parameters: https://travis-ci.org/maxplanck-ie/HiCExplorer/builds/210358414#L1803

File: galaxy/wrapper/hicFindTADs.xml (hunk 2 above)
Comment (vivekbhr) [inline galaxy/wrapper/hicFindTADs.xml]: `pvalue` is mis-typed..

File: galaxy/wrapper/hicFindTADs.xml (hunk 1 above)
Comment (fidelram) [inline galaxy/wrapper/hicFindTADs.xml]: I think 20MB is fine with github. There are not so many big files anyway.

Comment (fidelram) [inline galaxy/wrapper/hicFindTADs.xml]: is that file generated as part of the testing? if a low resolution matrix is used, a smaller zscore matrix for hicFindTads can be created.

**Comment by fidelram** [comment]
@bgruening I looked at the travis output but I do not understand what you mean.

**Comment by bgruening** [comment]
@fidelram this is the stacktrace:
```
 File "/home/travis/miniconda/bin/hicCorrectMatrix", line 7, in <module>
|  main()
|  File "/home/travis/miniconda/lib/python2.7/site-packages/hicexplorer/hicCorrectMatrix.py", line 657, in main
|  outlier_regions = filter_by_zscore(ma, args.filterThreshold[0], 1e6, perchr=args.perchr)
|  AttributeError: 'Namespace' object has no attribute 'perchr'
```
You added this in https://github.com/maxplanck-ie/HiCExplorer/commit/2a07c3cadcf3153bf9d5283926150d624fe0d4f3 11 days back. But not for the merge mode.

**Comment by fidelram** [comment]
@bgruening maybe I should remove the merge mode. That is sort of experimental and after using it I didn't recover many low coverage bins which was my goal.

**Comment by bgruening** [comment]
@fidelram makes sense!

**Comment by bgruening** [comment]
@fidelram @vivekbhr green :) !

**Pull request merged**